src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs

```diff
@@ -76,6 +83,56 @@ public async Task<ActionResult> ManageAatfs(ManageAatfsViewModel viewModel)
             }           
         }
 
+        [HttpGet]
+        public async Task<ActionResult> ManageContactDetails(Guid id)
+        {
+            using (var client = apiClient())
+            {
+                var contact = await client.SendAsync(User.GetAccessToken(), new GetAatfContact(id));
+
+                if (!contact.CanEditContactDetails)
+                {
+                    return new HttpForbiddenResult();
+                }
+
+                var viewModel = new AatfEditContactAddressViewModel()
+                {
+                    AatfId = id,
+                    ContactData = contact
+                };
+
+                viewModel.ContactData.AddressData.Countries = await client.SendAsync(User.GetAccessToken(), new GetCountries(false));
+                SetBreadcrumb();
+                return View(viewModel);
+            }
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> ManageContactDetails(AatfEditContactAddressViewModel viewModel)
+        {
+            if (ModelState.IsValid)
+            {
+                using (var client = apiClient())
+                {
+                    var request = requestCreator.ViewModelToRequest(viewModel);
+
+                    await client.SendAsync(User.GetAccessToken(), request);
+
+                    return RedirectToAction("Details", new { Id = viewModel.AatfId });
+                }
+            }
+
+            using (var client = apiClient())
+            {
+                viewModel.ContactData.AddressData.Countries = await client.SendAsync(User.GetAccessToken(), new GetCountries(false));
```

Comment:
Not sure we have done this elsewhere, but if we have to re get the countries list can we create a unit test to make sure the model property is set based from the api return

Reply:
I don't check this specifically, but I check that countries request is made?  AatfControllerTests line 211

Reply:
Cool, if you can check that the API call is made and the view model contains the value returns from the api...

Reply:
Added a test